### PR TITLE
[dagster-tableau] Implement data quality warning endpoint for data sources

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

Add `add_data_quality_warning_to_data_source` so that users can add a data quality warning to a Tableau data source with upstream assets are failing asset checks or materialization.

## How I Tested These Changes

BK with additional unit test

## Changelog

NOCHANGELOG

- [ ] `NEW` _(added new feature or capability)_
- [ ] `BUGFIX` _(fixed a bug)_
- [ ] `DOCS` _(added or updated documentation)_

### CODE REVIEW

**Comment by maximearmstrong** [comment]
* **#25043** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25043?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maximearmstrong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-tableau/dagster_tableau/resources.py [hunk 1]
```diff
@@ -82,6 +82,26 @@ def get_view(
...
 
+    def add_data_quality_warning_to_data_source(
+        self,
+        datasource_id: str,
+        warning_type: Optional[TSC.DQWItem.WarningType] = None,
+        message: Optional[str] = None,
+        active: Optional[bool] = None,
+        severe: Optional[bool] = None,
+    ) -> List[TSC.DQWItem]:
+        """Add a data quality warning to a data source."""
```
Comment (benpankow) [inline python_modules/libraries/dagster-tableau/dagster_tableau/resources.py:93]: nit: param docstr

**Review by benpankow — APPROVED** [review]
Seems reasonable to me. Where do we expect users to call this function? A check body? A sensor?

File: python_modules/libraries/dagster-tableau/dagster_tableau/resources.py (hunk 1 above)
Comment (maximearmstrong) [reply python_modules/libraries/dagster-tableau/dagster_tableau/resources.py]: Done in [225b57f](https://github.com/dagster-io/dagster/pull/25043/commits/225b57ff8a40aabab7bc6051079acf1b55e6b09a)

**Pull request merged**